File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
#include <unordered_set>
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> s;
        for (const int num : nums){
            if (s.contains(num)){
                return true;
            }
            s.insert(num);
        }
        return false;
    }
};

File: Data Structures & Algorithms/largest-unique-number/submission-4.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        freq = [0 for _ in range(1001)]

        for num in nums:
            freq[num] += 1
        
        for num in range(1000, -1, -1):
            if freq[num] == 1:
                return num
        return -1

        